api/types.proto

```diff
@@ -30,12 +30,27 @@ message Annotations {
 	repeated IndexEntry indices = 4 [(gogoproto.nullable) = false];
 }
 
+message Set {
```

Comment:
I'm really not a fan of this for a few reasons. I think this should just be a `repeated string` inside `GenericResource`:

- The internal representation shouldn't matter to the user
- `resource` now contains a mix of scalar and non-scalar (uint64 vs Set)
- `Set` is really weird (with `Void`)
- Using the API becomes hard

I think it would be easier if this is just:
```proto
oneof resource {
  uint64 discrete = 1;
  repeated string set = 2;
}
```